File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandDeleteArticle.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandDeleteArticle extends TelnetCommand {
    int _article_index;

    public BahamutCommandDeleteArticle(int articleIndex) {
        _article_index = articleIndex;
        Action = DeleteArticle;
    }

    public void execute(TelnetListPage aListPage) {
        if (_article_index > 0) {
            TelnetClient.getClient().sendStringToServer(_article_index + "\ndy");
        }
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[DeleteArticle][articleIndex=" + _article_index + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/command/BahamutCommandDeleteArticle.kt
package com.kota.Bahamut.command

import com.kota.Bahamut.listPage.TelnetListPage
import com.kota.Bahamut.listPage.TelnetListPageBlock
import com.kota.telnet.TelnetClient

class BahamutCommandDeleteArticle(var articleIndex: Int) : TelnetCommand() {
    init {
        action = BahamutCommandDef.Companion.DELETE_ARTICLE
    }

    override fun execute(telnetListPage: TelnetListPage) {
        if (articleIndex > 0) {
            TelnetClient.client?.sendStringToServer("$articleIndex\ndy")
        }
    }

    override fun executeFinished(telnetListPage: TelnetListPage, telnetListPageBlock: TelnetListPageBlock?) {
        isDone = true
    }

    override fun toString(): String {
        return "[DeleteArticle][articleIndex=$articleIndex]"
    }
}
